### PR TITLE
Fix #472: Respect logFrame/logNetwork options

### LINKED ISSUE #472: VirtualCluster configuration logFrame/logNetwork not respected.
**Describe the bug**

I want to see the frames sent to/from my clients to a virtual cluster.  I ought to be able to 
do this with the `logFrame` option in the configuration file. 

Setting this `true` shows this is ineffective.

**To Reproduce**
1. Deploy to minikube with `./scripts/run-with-strimzi.sh kubernetes-examples/portperbroker_plain`
2. Run `kubectl edit cm -n kafka kroxylicious-config` and set `logFrames` to true
3. Run `kubectl rollout restart -n kafka deployment kroxylicious-proxy`
4. Run an example kafka command 
5. Run `kubectl logs -f deployment/kroxylicious-proxy -n kafka` 

**Actual behavior**
Empty log

**Expected behavior**
Logs such contain frames interaction.

**YAML files and logs**

Attach or copy paste your kroxylicious configuration YAML (with sensitive details
like broker address redacted).
Attach or copy and paste relevant logs.

**Additional context**

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Bugfix

### Description

Respect logFrame/logNetwork configuration options.

* declare loggers in the log4j2 configuration file corresponding to the frame and network loggers
* also define two environment variables that make it easy to override useful loggers to aid problem resolution.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [x] Reference relevant issue(s) and close them after merging
- [x] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java [hunk 1]
```diff
@@ -89,7 +90,7 @@ public KafkaProxy(Configuration config) {
      * @return This proxy.
      */
     public KafkaProxy startup() throws InterruptedException {
-        LOGGER.info("Kroxylicious starting");
+        STARTUP_SHUTDOWN_LOGGER.info("Kroxylicious starting");
```
Comment (k-wall) [inline kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java:93]: I want an Asciiart kroxy as a startup message.

File: kroxylicious/src/main/resources/log4j2.properties [hunk 2]
```diff
@@ -11,11 +11,23 @@ appender.console.name = STDOUT
...
 rootLogger.appenderRef.console.ref = STDOUT
 rootLogger.additivity = false
 
-#logger.netty.name = io.netty
-#logger.netty.level = TRACE
-#logger.netty.additivity = false
+# The DownstreamNetworkLogger and DownstreamFrameLogger loggers are required to allow
+# the virtual cluster logNetwork/logFrames feature to work.  The log level needs to be in sync
+# with the logger level used in the code.
+logger.DownstreamNetworkLogger.name = io.kroxylicious.proxy.internal.DownstreamNetworkLogger
```
Comment (SamBarker) [inline kroxylicious/src/main/resources/log4j2.properties:22]: Does the existence of `Downstream*Logger` imply that we should also have a `Upstream*Logger` ?

File: kroxylicious/src/main/resources/log4j2.properties [hunk 3]
```diff
@@ -11,11 +11,23 @@ appender.console.name = STDOUT
...
+logger.DownstreamNetworkLogger.level = INFO
+
+logger.DownstreamFrameLogger.name = io.kroxylicious.proxy.internal.DownstreamFrameLogger
+logger.DownstreamFrameLogger.level = INFO
+
+logger.StartupShutdownLogger.name = io.kroxylicious.proxy.internal.StartupShutdownLogger
+logger.StartupShutdownLogger.level = INFO
+
+logger.test.name = io.kroxylicious.proxy.KafkaProxy
+logger.test.level = ${env:KROXYLICIOUS_LOG_LEVEL:-${env:KROXYLICIOUS_ROOT_LOG_LEVEL:-WARN}}
```
Comment (SamBarker) [inline kroxylicious/src/main/resources/log4j2.properties:32]: Thinking out loud should *tests* default to more verbose logging? Are we more likely to catch issues etc when running tests with logging turned up? 

A quiet build is also a desirable property. I'm tempted to say the build server should run with a higher level of logging on by default so we can diagnose issues there (which this change facilitates nicely)?

File: kroxylicious/src/main/resources/log4j2.properties (hunk 2 above)
Comment (SamBarker) [reply kroxylicious/src/main/resources/log4j2.properties:22]: Yeah, so checking why `io.kroxylicious.proxy.internal.DownstreamNetworkLogger` doesn't appear in the diff outside of the log file confirms that `io.kroxylicious.proxy.internal.UpstreamNetworkLogger` is a thing so can we add that here as well?

Comment (SamBarker) [inline kroxylicious/src/main/resources/log4j2.properties:22]: Should we also be configuring a separate appender for network and or frame logging? So we can default it to file?

**Review by SamBarker — APPROVED** [review]

File: kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java (hunk 1 above)
Comment (SamBarker) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java:93]: does https://github.com/k-wall/kroxylicious/pull/3 fit the bill?

Comment (k-wall) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java]: Adopted :)

File: kroxylicious/src/main/java/io/kroxylicious/proxy/BannerLogger.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        @Override
+        public Stream<String> get() {
+            final InputStream resourceStream = getClass().getClassLoader().getResourceAsStream(resourceName);
+            if (Objects.isNull(resourceStream)) {
+                return Stream.empty();
+            }
+            else {
+                final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(resourceStream));
```
[the remark this answers is not in the document]
Comment (k-wall) [reply kroxylicious/src/main/java/io/kroxylicious/proxy/BannerLogger.java:80]: Fixed.

File: kroxylicious/src/main/resources/log4j2.properties (hunk 2 above)
Comment (SamBarker) [reply kroxylicious/src/main/resources/log4j2.properties:22]: Yeah, I was wondering about the K8s log everything to stdout model, but I suspect the majority of the time we want frame and network logging is in dev/local envs and keeping them in a separate file feels tempting to me for that usage.

**Pull request merged**